lib/spack/spack/directory_layout.py

```diff
@@ -90,14 +90,28 @@ def path_for_spec(self, spec):
         assert(not path.startswith(self.root))
         return os.path.join(self.root, path)
 
-    def remove_install_directory(self, spec):
+    def remove_install_directory(self, spec, deprecated=False):
         """Removes a prefix and any empty parent directories from the root.
            Raised RemoveFailedError if something goes wrong.
         """
         path = self.path_for_spec(spec)
         assert(path.startswith(self.root))
 
-        if os.path.exists(path):
+        if deprecated:
+            if os.path.exists(path):
+                try:
+                    dep_file_name = self.deprecated_file_name(spec)
```

Comment:
(minor) I think it would be slightly clearer if `deprecated_file_path` allowed not providing `replacement_spec` in which case it would do this calculation (and the docstring/comments could mention that you can construct the path for an already-deprecated spec by following the prefix symlink).